src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+
+public class Intake extends SubsystemBase {
+    private final DigitalInput preIntakeSensor = new DigitalInput(0);
+    private final DigitalInput intakeSensor = new DigitalInput(1);
+    private final DigitalInput shooterSensor = new DigitalInput(2);
+
+    private final TalonFX intakeMotor = new TalonFX(0);
+
+    boolean status = intakeMotor.isAlive();
+    // Will change intakeMotorVelocity
+    double intakeMotorVelocity = 1;
+    private VelocityVoltage velocity = new VelocityVoltage(intakeMotorVelocity);
+
+    enum IntakeState {
+        IDLE, // No motors are moving, no note is inside the mechanism
+        INTAKING, // Motors are moving, note is not yet where we need it to be
+        HOLDING, // No motors are moving, note is where it needs to be and is contained in the robot
+        EJECTING; // Motors are moving, note is being moved into the gears
+    }
+
+    public static IntakeState CurrentState = IntakeState.IDLE;
+
+    public Intake() {
+        // SmartDashboard displays to test sensors
+        SmartDashboard.putBoolean("Motor Functional?", status);
+        SmartDashboard.putNumber("Intake Motor Speed (RPM)", intakeMotorVelocity);
+    }
+
+    @Override
+    public void periodic() {
+        switch (CurrentState) {
+            case IDLE:
+                if (preIntakeSensor.get() == false) {
```

Comment:
When you access the same variable multiple times, you should extract it to a variable at the top of the period method, outside of the switch/case. The reason why is because the sensor value can change in the middle of this method and that will be hard to think through logically.

At the top, (as an example) I'd add:
```java
boolean noteAtPreIntakeSensor = ! preIntakeSensor.get();
boolean noteAtIntakeSensor = ! intakeSensor.get();
boolean noteAtShooterSensor = ! shooterSensor.get();
```

They also make it a little easier to determine if `false` from get means "the note is there" or if `true` means "the note is there". If you're wrong about it, you only have to change it at the variable definition by adding or removing the `!` instead of inverting all of your logic.

The other thing I'd say is make sure your if statements use `!` properly. For example, you can just write:
```java
if(! noteAtPreIntakeSensor)
```

or down below on line 51:
```java
if(noteAtPreIntakeSensor && noteAtIntakeSensor)
```